Tests/ApodiniNetworkingTests/ApodiniNetworkingTests.swift

```diff
@@ -69,4 +70,44 @@ class ApodiniNetworkingTests: XCTestCase {
             effectivePath: "/v[v1]/v[greet]/**[wm]"
         )
     }
+    
+    
+    func testQueryParamDecoding() throws {
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(String.self, from: "abcdef"), "abcdef")
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Int.self, from: "1234"), 1234)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Int.self, from: "-1234"), -1234)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Double.self, from: "1234"), 1234)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Double.self, from: "1234.5678"), 1234.5678)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Double.self, from: "-1234.5678"), -1234.5678)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Float.self, from: "1234"), 1234)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Float.self, from: "1234.5678"), 1234.5678)
+        XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Float.self, from: "-1234.5678"), -1234.5678)
+        XCTAssertThrowsError(try URLQueryParameterValueDecoder().decode(Int.self, from: "abcdef"))
+        for trueBoolInput in ["true", "True", "TRUE", "yes", "Yes", "YES", "1"] {
+            XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Bool.self, from: trueBoolInput), true)
+        }
+        for falseBoolInput in ["false", "False", "FALSE", "no", "No", "NO", "0"] {
+            XCTAssertEqual(try URLQueryParameterValueDecoder().decode(Bool.self, from: falseBoolInput), false)
+        }
+        for invalidBoolInput in ["2", "3", "4", "a", "b", "c"] { // obviously not an exhaustive list...
+            XCTAssertThrowsError(try URLQueryParameterValueDecoder().decode(Bool.self, from: invalidBoolInput))
+        }
+    }
+    
+    
+    func testQueryParamDateDecoding() throws {
+        func imp(_ strategy: DateDecodingStrategy, input: String) throws -> Date {
+            try URLQueryParameterValueDecoder(dateDecodingStrategy: strategy).decode(Date.self, from: input)
+        }
+        for _ in 0...1000 {
```

Comment:
Is that actually effective/necessary?

Reply:
There were some inconsistencies where the `Date` decoding would randomly fail (due to how Foundation deals with timestamps; this is also the reason why I'm comparing the dates' `timeIntervalSince1970`, instead of comparing the dates directly), so I added the loop to repeat the test a bunch of times. This increases the possibility of us detecting a failure.